haystack/utils/openai_utils.py

```diff
@@ -87,16 +87,25 @@ def _openai_text_completion_tokenization_details(model_name: str):
 @retry_with_exponential_backoff(
     backoff_in_seconds=OPENAI_BACKOFF, max_retries=OPENAI_MAX_RETRIES, errors=(OpenAIRateLimitError, OpenAIError)
 )
-def openai_request(url: str, headers: Dict, payload: Dict, timeout: Union[float, Tuple[float, float]] = OPENAI_TIMEOUT):
+def openai_request(
+    url: str,
+    headers: Dict,
+    payload: Dict,
+    timeout: Union[float, Tuple[float, float]] = OPENAI_TIMEOUT,
+    load_response: Optional[bool] = True,
+    **kwargs,
+):
     """Make a request to the OpenAI API given a `url`, `headers`, `payload`, and `timeout`.
 
     :param url: The URL of the OpenAI API.
     :param headers: Dictionary of HTTP Headers to send with the :class:`Request`.
     :param payload: The payload to send with the request.
     :param timeout: The timeout length of the request. The default is 30s.
+    :param load_response: Whether to load the response as JSON. The default is True.
```

Comment:
nitpick: I had to read the code to understand what `load_response` means. Let's call it something like  `json_response: bool`. What I found confusing is really only the verb to 'load', so I think any synonym (decode, read, etc...) would work too.

Reply:
Yeah, I didn't like it either. Thanks for suggesting it, I'll change it. I need to write that common REST invocation layer at the first opportunity. I don't like that we have this OpenAI call going to a util method. A unified REST layer with a map of exceptions, retry options and timeouts should clean up this code a lot